docs/ci/ci-automated-release-workflow.md

```diff
@@ -0,0 +1,120 @@
+3. Optionally trigger an E2E test against that pull request,
+4. Wait for all pull request CI checks to complete,
+5. Manually request two maintainers to approve the pull request, and
+6. Manually create a release from the GitHub UI using that new branch
+
+This entire process takes at least 10 minutes of manual work, plus however long it takes for the pull request's checks to complete. (In some repositories, like the core repo, this can take 2+ hours.)
+
+Going forward, we should automate these release processes so that contributors and maintainers can focus more on development work and less on creating actual releases.
+
+## Automation Workflow
+
+### Brief Overview of the Automation
+
+The automation logic described in this dev-doc will be published in the form of an in-house GitHub action called `create-automated-release` and it will therefore be callable from any workflow file. Thus, it is strongly recommended that each repository maintainer creates a `.github/workflows/automated-release.yml` workflow file to call this `create-automated-release` GitHub action from.
+
+### Goals of the Automation
+
+The automated process should be:
+
+1. Configurable so that any library maintainer can configure the automation to meet their specific repository's needs
+2. Schedulable so that releases are generated according to a specific cadence
+
+Scheduled releases are generally important for setting expectations around release cadences, but they are by no means required for every library and may not even be applicable to some in this GitHub organization.
+
+### Configurable Components
+
+As mentioned above, there are configurable components within this release process automation. The diagram in the next section references two configurable components:
+
+#### Trigger Schedule
+
+The trigger schedule defines the day and time (in UTC) when the release process will kick off.  This schedule can be disabled if desired, and maintainers can trigger the release process manually when needed instead.
+
+As mentioned in the brief overview of the automation, this in-house `create-automated-release` action will be callable from any workflow file. Thus, each library maintainer who wants to create a scheduled release should first create a  `.github/workflows/automated-release.yml` workflow file in their repository and define a schedule for it. For example:
+
+```yaml
+name: Create Y-Stream Release
+
+on:
+  schedule:
+    - cron: '30 1 1,15 * *' # Triggers at 1:30am UTC every 2 weeks on the 1st and the 15th day of each month
+```
+
+#### Custom List of Dependencies to Cap
+
+With each release, some library maintainers may want to cap the version of certain dependencies within their `requirements.txt` file as well as specify the desired upper cap for each one.
+
+The list of dependencies to cap should be provided in a file called `automated-release-config.yml`. This configuration file may be expanded in the future to accommodate more configurations as needed. Example:
```

Comment:
@courtneypacheco Would this `automated-release-config.yml` file be per `y` stream release ? Is there a way we could add a checkpoint to verify in the beginning of the workflow to make sure this file exists and if it does, make sure it exists with the necessary details and format ? We could skip the secondary validation, if the file doesn't exist, which seems to be a possibility based on your workflow diagram.

Reply:
Also would it be a good idea to version this file within the `y` stream release, if things change on us ? In other words, do we want this file to be immutable once the release process kicks off.